neon_llm_vllm/vllm.py

```diff
@@ -196,8 +194,15 @@ def parse_persona_dict(self, persona: dict) -> dict:
         # get model
         model_name = description.split(",")[0].strip()
         if model_name not in self.models:
-            # fall back to first model
-            model_name = list(self.models.keys())[0]
+            for model in self.models:
+                if model.split('@')[0] == model_name.split('@')[0]:
+                    model_name = model
+                    break
```

Comment:
Do we want to support gradio notation with this
What is the reason? Looks like it just redefines splitter to support both `,` or `@`

Reply:
Lover i see the code `persona_name = persona['description'].split(',')[0].strip()`
That makes this question appear again

Reply:
I mostly figured this out experimentally; here I saw `model` and `model_name` look like `BrainForge/NeonLLM@2025-02-05`, so I used those to compare.

The full `description` string looks to be `<model_repo>@<model_revision>,<persona_name>` which should be fine..

Reply:
It looks like this logic is dublicate of `if model_name not in self.models:`

Reply:
Oh, you want to reduce amounth of reboots by ignoring revision in matching?

Reply:
> Oh, you want to reduce amounth of reboots by ignoring revision in matching?

That is part of it; I also want to try and fall-back to something close to the request if possible since it isn't obvious to the user when a persona other than the one they requested is selected to respond. Restarting VLLM also doesn't appear to update the personas in Klat, but I will investigate that separately.

Reply:
I agree, if we need to add falback(because of klat) your works better